packages/lambdas/src/fhir-to-bundle.ts

```diff
@@ -14,20 +15,24 @@ capture.init();
 // Set by us
 const apiUrl = getEnvOrFail("API_URL");
 const bucketName = getEnvOrFail("BUCKET_NAME");
+const region = getEnvOrFail("AWS_REGION");
+const postHogSecretName = getEnvOrFail("POST_HOG_API_KEY_SECRET");
 
 export async function handler(
   params: ConsolidatedSnapshotRequestSync | ConsolidatedSnapshotRequestAsync
 ): Promise<ConsolidatedSnapshotResponse | void> {
+  const postHogApiKey = await getSecretValue(postHogSecretName, region);
   const { patient, requestId, resources, dateFrom, dateTo } = params;
   const conversionType = params.isAsync ? params.conversionType : undefined;
   const { log } = out(`cx ${patient.cxId}, patient ${patient.id}, req ${requestId}`);
+
   try {
     log(
       `Running with dateFrom: ${dateFrom}, dateTo: ${dateTo}, conversionType: ${conversionType}` +
         `, resources: ${resources}}`
     );
     const conn = new ConsolidatedSnapshotConnectorLocal(bucketName, apiUrl);
-    const result = await conn.execute(params);
+    const result = await conn.execute({ ...params, postHogApiKey });
```

Comment:
Because ConsolidatedSnapshotConnectorLocal can be used both locally and in lambda i think it may be better to return the results we want to track here and then send the posthog analyticsAsync message

Reply:
Another thing as well instead of sending 3 separate events for each step of cleaning data lets try to maybe send one with all the necessary details (can help save on costs and seems reasonable to consolidated them into one event)